code/modules/mod/modules/modules_general.dm

```diff
@@ -95,38 +95,43 @@
 	var/stabilizers = FALSE
 	/// Do we give the wearer a speed buff.
 	var/full_speed = FALSE
-	/// The ion trail particles left after the jetpack.
-	var/datum/effect_system/trail_follow/ion/grav_allowed/ion_trail
+	var/datum/callback/get_mover
+	var/datum/callback/check_on_move
 
 /obj/item/mod/module/jetpack/Initialize(mapload)
 	. = ..()
-	ion_trail = new()
-	ion_trail.auto_process = FALSE
-	ion_trail.set_up(src)
+	get_mover = CALLBACK(src, .proc/get_user)
+	check_on_move = CALLBACK(src, .proc/allow_thrust)
+	refresh_jetpack()
 
 /obj/item/mod/module/jetpack/Destroy()
-	QDEL_NULL(ion_trail)
+	get_mover = null
+	check_on_move = null
 	return ..()
 
+/obj/item/mod/module/jetpack/proc/refresh_jetpack()
+	AddComponent(/datum/component/jetpack, stabilizers, COMSIG_JETMODULE_ACTIVATED, COMSIG_JETMODULE_DEACTIVATED, JETMODULE_ACTIVATION_FAILED, get_mover, check_on_move, /datum/effect_system/trail_follow/ion/grav_allowed)
+
+/obj/item/mod/module/jetpack/proc/set_stabilizers(new_stabilizers)
+	if(stabilizers == new_stabilizers)
+		return
+	stabilizers = new_stabilizers
+	refresh_jetpack()
+
 /obj/item/mod/module/jetpack/on_activation()
 	. = ..()
 	if(!.)
 		return
-	ion_trail.start()
-	RegisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED, .proc/move_react)
-	RegisterSignal(mod.wearer, COMSIG_MOVABLE_PRE_MOVE, .proc/pre_move_react)
-	RegisterSignal(mod.wearer, COMSIG_MOVABLE_SPACEMOVE, .proc/spacemove_react)
+	if(SEND_SIGNAL(src, COMSIG_JETMODULE_ACTIVATED) & JETMODULE_ACTIVATION_FAILED)
+		return
```

Comment:
do we need these? modsuits already have a signal for activation

Reply:
Would it make sense to add blocking to that proc? The return value means something different to it then the flag being set here, right?